src/components/layerpublisher/LayerPublisher.js

```diff
@@ -24,43 +43,60 @@ function info(fileName) {
   PubSub.publish(STATUS_INFO, `Loaded ${fileName}.`);
 }
 
+function fetchDataFromDZI(layerType, tileSource) {
+  return fetch(tileSource).then(response => response.text())
+    .then(str => (new window.DOMParser()).parseFromString(str, 'text/xml'))
+    .then((layer) => {
+      const layerData = {};
+      if (layer.getElementsByTagName('Image')[0].attributes.Overlap.value !== 0) {
+        console.error('Overlap is nonzero - image will have artifacts');
+        throw new Error('Tiles have overlap');
+      }
+      layerData.layerType = layerType;
+      layerData.tileSource = tileSource.substring(0, tileSource.lastIndexOf('/'));
```

Comment:
Comment explaining what's going on here... Will there be a filename following the slash? Could be confusing that `tileSource` has different meanings depending on context.

Reply:
Fair